pkg/controller/jobcontrol.go

```diff
@@ -207,34 +207,16 @@ func (c *jobControl) ControlJobs(
 			currentJob = job
 		}
 		if jobOwnerGeneration(job) == owner.GetGeneration() {
-			generationJob = job
+			if isActive(job) {
+				activeJob = job
+			} else if isSuccessful(job) {
+				successfulJob = job
+			}
```

Comment:
Are we intentionally not deleting failed jobs for the current generation? All jobs for a given generation will persist until the generation is updated.

Reply:
I was thinking that we wouldn't for the current generation so you could get an idea of why they're failing. Since we create a new job every 3 tries, and suppose that you have a job that takes 1 hr to fail ... and you're in the middle of a new job you just started. It'd be nice to have the previous job so you can see what happened.

Reply:
That makes sense. I was just verifying that it was intentional. At some point, we'll need to clean up old jobs (failed and succeeded), but that is a problem that we can address later.

I guess one concern that I have is that if there is a job that is perpetually failing, then we will use up a lot of resources keeping around the failed jobs and their pods.

Reply:
We could trim the number of jobs we keep around. Is it ok if we do it as a follow up ?

Reply:
Yes.